internal/mesh/internal/controllers/routes/generate.go

```diff
@@ -68,6 +70,8 @@ func compile(
 	}
 	parentServiceID = parentServiceDec.Resource.Id // get ULID out of it
 
+	boundRefCollector.AddRefOrID(parentServiceRef)
```

Comment:
I only add it to the collector if the following are all true:
- we load the resource for some reason
- the resource is **found**
- we decided to use the information in that resource to produce `ComputedRoutes`

Those seem like suitable criteria to then use the persisted bound references to then later purge influence correctly.
-

Reply:
I decided to add all of the relevant references to the generated resource, if only for diagnostic reasons.

I _think_ only the xRoute mappers need to be updated due to their unique indexing strategy (meaningless name, ref in body).